File: common/src/main/java/org/valkyrienskies/clockwork/content/contraptions/fan/EncasedFanData.java
package org.valkyrienskies.clockwork.content.contraptions.fan;

import org.joml.Vector3dc;

public class EncasedFanData {
    public final Vector3dc fanPos;
    public final Vector3dc fanDir;

    public double fanSpeed;

    public EncasedFanData(Vector3dc fanPos, Vector3dc fanDir, double fanSpeed) {
        this.fanPos = fanPos;
        this.fanDir = fanDir;
        this.fanSpeed = fanSpeed;
    }
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/content/propulsion/singleton/fan/EncasedFanData.kt
package org.valkyrienskies.clockwork.content.propulsion.singleton.fan

import com.fasterxml.jackson.annotation.JsonAutoDetect
import org.joml.Vector3dc

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
class EncasedFanData(val fanPos: Vector3dc, val fanDir: Vector3dc, var fanSpeed: Double)